### PR TITLE
Get flowgraph nodes

### PR DESCRIPTION
**Pull request by troibe**
**Why?**
When working more with the flowgraph it was a bit unclear at times what is a node vs step in existing code.
Additonally this should make it more clear at which points the steplist and indexlist are considered in the flowgraph.
Last minor benefit is that it saves one level of indentation.

**What?**
Introduce a function that gets all nodes `(step, index)` in the flowgraph and use it in existing code.

**Testing:**
[Remote server run](https://github.com/zeroasiccorp/scserver/actions/runs/6096289187/job/16542015595)

### CODE REVIEW

File: siliconcompiler/core.py [hunk 1]
```diff
@@ -1688,43 +1688,35 @@ def check_manifest(self):
...
+                self.logger.error(f'{step}{index} relies on {in_step}{in_index}, '
+                                  'but this task has not been run and is not in the '
+                                  'current steplist.')
+                error = True
 
         # 2. Check library names
         libraries = set()
         for val, step, index in self.schema._getvals('asic', 'logiclib'):
-            if step in steplist and index in indexlist[step]:
+            if step in steplist and (not indexlist or in_index in indexlist):
```
Comment (gadfort) [inline siliconcompiler/core.py:1719]: why not check for `(step,index) in nodes` instead of using the index list

File: tests/core/test_graph.py [hunk 2]
```diff
@@ -64,6 +64,25 @@ def test_graph():
...
+    chip.node(flow, 'premin', fake_out, index=1)
+    chip.node(flow, 'domin', minimum)
+    chip.node(flow, 'postmin', fake_in)
+
+    chip.edge(flow, 'premin', 'domin', tail_index=0)
+    chip.edge(flow, 'premin', 'domin', tail_index=1)
+    chip.edge(flow, 'domin', 'postmin')
+
+    assert chip._get_flowgraph_nodes(flow) == \
+        [('premin', '0'), ('premin', '1'), ('domin', '0'), ('postmin', '0')]
```
Comment (gadfort) [inline tests/core/test_graph.py:83]: this this order always true?

File: siliconcompiler/core.py (hunk 1 above)
Comment (gadfort) [reply siliconcompiler/core.py:1719]: it seems like the get_flowgraph_nodes will return based on the indexlist, so why not just use that where you have checks for indexlist?

Comment (troibe) [reply siliconcompiler/core.py:1719]: Just replaced it there.
At some other points there is some logic around the lines of: `if` at least one index of this step or all indices of this step
There I didn't find a prettier replacement yet.

File: tests/core/test_graph.py (hunk 2 above)
Comment (troibe) [reply tests/core/test_graph.py:83]: For now yes it is the order from the schema and it is a list so it is a fixed order.

**Review by gadfort — APPROVED** [review]

**Pull request merged**